### PR TITLE
Added extra functions

### PR DESCRIPTION
**Pull request by mgmoens**
**Changes introduced by this pull request**:
Added log, sin, cos and tan functions to calc.
**TODOs**: [what is not done yet.]
Fix some crappy code construction I did
**Fixes**: [what issues this fixes.]
https://github.com/redox-os/calc/issues/1

### CODE REVIEW

**Comment by mgmoens** [comment]
Imma just mention you since you are the "maintainer" and you aren't watching this.
@huntergoldstein

**Comment by hgoldstein** [comment]
The Travis build failed due to not being formatted correctly; `cargo fmt` ought to do it. Taking a look now (and thanks for tagging me)!

File: src/lib.rs [hunk 1]
```diff
@@ -497,6 +559,40 @@ fn g_expr(token_list: &[Token]) -> Result<IntermediateResult, CalcError> {
...
+                            Token::CloseParen => {
+                                let mut val = 0.0; //TODO: not declare this val here like the warningwants
+                                match token_list[0] {
+                                    Token::Cosine => val = ir.value.cos(),
+                                    Token::Sine => val = ir.value.sin(),
+                                    Token::Tangent => val = ir.value.tan(),
+                                    Token::Logarithm => {
+                                        val = ir.value.log(10.0)
+                                    }
+                                    _ => val = 0.0, //TODO: maybe get this outta the way too
```
Comment (hgoldstein) [inline src/lib.rs]: This ought to be an `unreachable!()` versus silently returning zero.

**Review by hgoldstein — COMMENTED** [review]
My only suggestion is that you might have an easier time allowing functions to be invoked without parens:
`log 10` versus `log(10)`. It should be straightforward.

I also don't think implementing `OperatorMatch` for arrays of increasing size is a good idea. However this can be revisited with #2 I think.

File: src/lib.rs [hunk 2]
```diff
@@ -73,24 +81,29 @@ pub enum CalcError {
...
+                ["calc: unexpected ", kind, " token: ", &token].concat()
             }
-            UnexpectedEndOfInput => write!(f, "unexpected end of input"),
-            UnmatchedParenthesis => write!(f, "unmatched patenthesis"),
+            UnexpectedEndOfInput => "calc: unexpected end of input".into(),
+            UnmatchedParenthesis => "calc: unmatched parenthesis".into(),
+            NoFunctionArgument => "calc: no bracketed function argument".into(),
         }
     }
 }
```
Comment (hgoldstein) [inline src/lib.rs]: I'm not entirely sure how this happened, but this is also causing Travis to fail: `bin.rs` expects `CalcError` to implement `Display` but it only has `To<String>`.

**Comment by mgmoens** [comment]
This should work :) I hadn't pulled from the master since yesterday or so and I missed a commit. This kinda screwed up a couple of things.

**Pull request merged**